rebasehelper/tests/test_specfile.py

```diff
@@ -199,6 +199,34 @@ def test_find_archive_target_in_prep(self, spec_object):
         target = spec_object.find_archive_target_in_prep('misc.zip')
         assert target == 'test-1.0.2/misc'
 
+    def test_write_updated_patches(self, mock_spec_object_generator):
+        content = """Patch0:    0.patch
+Patch1:    1.patch
+
+%patchlist
+2.patch
+3.patch
+
+%prep
+%patch0 -p0
+%patch1 -p1
+%patch2 -p0
+%patch3 -p0
+"""
+        expected = """Patch0:    0.patch
+
+%patchlist
+2.patch
+
+%prep
+%patch0 -p0
+%patch1 -p0
+"""
+        spec = mock_spec_object_generator(content)
+        spec.removed_patches = []
+        spec.write_updated_patches({'deleted': ['1.patch', '3.patch']}, False)
```

Comment:
This test ignores inapplicable patches, but that's exactly the part I suspect to be a bit buggy.

Reply:
Updated the test to also disable an inapplicable patch. Seems to work correctly